File: backend/NodeJs/src/shared/entities/TimeOnlineStatus.ts
import {
  Column,
  Entity,
  Index,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Players } from './Players';

@Index('player_id', ['playerId'], {})
@Entity('time_online_status')
export class TimeOnlineStatus {
  @PrimaryGeneratedColumn({ type: 'int', name: 'id' })
  id: number;

  @Column('int', { name: 'player_id', nullable: true })
  playerId: number | null;

  @Column('int', { name: 'status', nullable: true, default: () => "'0'" })
  status: number | null;

  @Column('timestamp', {
    name: 'timestamp',
    nullable: true,
    default: () => 'CURRENT_TIMESTAMP',
  })
  timestamp: Date | null;

  @ManyToOne(() => Players, (players) => players.timeOnlineStatuses, {
    onDelete: 'CASCADE',
    onUpdate: 'NO ACTION',
  })
  @JoinColumn([{ name: 'player_id', referencedColumnName: 'id' }])
  player: Players;
}

File: backend/ASP.NET/SurfGxds/Models/TimeOnlineStatus.cs
using System;
using System.Collections.Generic;

namespace SurfGxds.Models
{
    public partial class TimeOnlineStatus
    {
        public int Id { get; set; }
        public int? PlayerId { get; set; }
        public int? Status { get; set; }
        public DateTime? Timestamp { get; set; }

        public virtual Player? Player { get; set; }
    }
}
